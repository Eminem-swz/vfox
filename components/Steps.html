<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>Steps 步骤条 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.82adce24.css">
    <link rel="modulepreload" href="/vfox/assets/Home.472bd3cf.js">
    <link rel="modulepreload" href="/vfox/assets/app.9dc78301.js">
    <link rel="modulepreload" href="/vfox/assets/components_Steps.md.9ca726e6.lean.js">
    
    <meta name="twitter:title" content="Steps 步骤条 | Vfox for Vue3">
  <meta property="og:title" content="Steps 步骤条 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-675d8756><div class="sidebar-button" data-v-675d8756><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-675d8756 data-v-4a583abe><!----> Vfox for Vue3</a><div class="flex-grow" data-v-675d8756></div><div class="nav" data-v-675d8756><nav class="nav-links" data-v-675d8756 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-83e92a68><nav class="nav-links nav" data-v-83e92a68 data-v-eab3edfe><!--[--><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/" data-v-b8818f8c>指南 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/design/" data-v-b8818f8c>设计 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item active" href="/vfox/components/Button" data-v-b8818f8c>组件 <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-link" data-v-eab3edfe data-v-b8818f8c><a class="item" href="/vfox/apis/Toast" data-v-b8818f8c>API <!----></a></div></div><div class="item" data-v-eab3edfe><div class="nav-dropdown-link" data-v-eab3edfe data-v-56bf3a3f><button class="button" data-v-56bf3a3f><span class="button-text" data-v-56bf3a3f>选择版本</span><span class="right button-arrow" data-v-56bf3a3f></span></button><ul class="dialog" data-v-56bf3a3f><!--[--><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item isExternal" href="https://cdn.fox2.cn/1.x/docs/" target="_blank" rel="noopener noreferrer" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>1.x</span><span class="icon" data-v-bbc27490><svg class="icon outbound" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" x="0px" y="0px" viewbox="0 0 100 100" width="15" height="15" data-v-bbc27490><path fill="currentColor" d="M18.8,85.1h56l0,0c2.2,0,4-1.8,4-4v-32h-8v28h-48v-48h28v-8h-32l0,0c-2.2,0-4,1.8-4,4v56C14.8,83.3,16.6,85.1,18.8,85.1z"></path><polygon fill="currentColor" points="45.7,48.7 51.3,54.3 77.2,28.5 77.2,37.2 85.2,37.2 85.2,14.9 62.8,14.9 62.8,22.9 71.5,22.9"></polygon></svg></span></a></div></li><li class="dialog-item" data-v-56bf3a3f><div class="nav-dropdown-link-item" data-v-56bf3a3f data-v-bbc27490><a class="item active" href="/vfox/" data-v-bbc27490><span class="arrow" data-v-bbc27490></span><span class="text" data-v-bbc27490>2.x</span><span class="icon" data-v-bbc27490><!----></span></a></div></li><!--]--></ul></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-83e92a68><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/Steps">Steps</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#steps-props">Steps Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#steps-slots">Steps Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#默认（-default）">默认（#default）</a><!----></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="#step-props">Step Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#step-slots">Step Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#内容（-default）">内容（#default）</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#标题（-title）">标题（#title）</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#步骤标（-step）">步骤标（#step）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CircleProgress">CircleProgress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-7eddb2c4><div class="container" data-v-7eddb2c4><!--[--><!--[--><!----><!--]--><!--[--><div><iframe style="width:375px;height:667px;margin-top:1.5rem;" src="https://godxiaoji.github.io/vfox/demo/#/Steps" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-7eddb2c4><div><h1 id="steps-步骤条" tabindex="-1">Steps 步骤条 <a class="header-anchor" href="#steps-步骤条" aria-hidden="true">#</a></h1><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-"><pre><code>import { Steps, Step } from &#39;vfox&#39;
</code></pre></div><p>具体的引入方式可以参考<a href="./../guide/import.html">引入组件</a>。</p><h2 id="steps-props" tabindex="-1">Steps Props <a class="header-anchor" href="#steps-props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>v-model:active-index</td><td>number</td><td>0</td><td>否</td><td>当前步骤对应的索引值</td></tr><tr><td>dot</td><td>boolean</td><td>false</td><td>否</td><td>是否开启小点样式</td></tr></tbody></table><h2 id="steps-slots" tabindex="-1">Steps Slots <a class="header-anchor" href="#steps-slots" aria-hidden="true">#</a></h2><h3 id="默认（-default）" tabindex="-1">默认（#default） <a class="header-anchor" href="#默认（-default）" aria-hidden="true">#</a></h3><p>注：其中只可放置 <a href="./Steps.html#step-props">Step</a> 组件，否则会导致未定义的行为。</p><div class="language-"><pre><code>&lt;fx-steps&gt;
  &lt;fx-step title=&quot;成功获得0.01元收益&quot;&gt;搞半天就这点？&lt;/fx-step&gt;
  &lt;fx-step title=&quot;十天后到账&quot;&gt;0.01元还要十天到账？&lt;/fx-step&gt;
  &lt;fx-step title=&quot;爱要不要&quot;&gt;不要了，滚。&lt;/fx-step&gt;
&lt;/fx-steps&gt;
</code></pre></div><h2 id="step-props" tabindex="-1">Step Props <a class="header-anchor" href="#step-props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>title</td><td>string</td><td></td><td>否</td><td>步骤子项标题</td></tr></tbody></table><h2 id="step-slots" tabindex="-1">Step Slots <a class="header-anchor" href="#step-slots" aria-hidden="true">#</a></h2><h3 id="内容（-default）" tabindex="-1">内容（#default） <a class="header-anchor" href="#内容（-default）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-step title=&quot;标题&quot;&gt;
  自定义内容
&lt;/fx-step&gt;
</code></pre></div><h3 id="标题（-title）" tabindex="-1">标题（#title） <a class="header-anchor" href="#标题（-title）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-step&gt;
  &lt;template #title&gt;
  【珠海市】【珠海一部】快递小哥正在派件（&lt;a href=&quot;tel:10000&quot;&gt;10000&lt;/a&gt;）
  &lt;/template&gt;
  2021-04-13 11:22:16
&lt;/fx-step&gt;
</code></pre></div><p>PS：优先级高于 Props <code>title</code>。</p><h3 id="步骤标（-step）" tabindex="-1">步骤标（#step） <a class="header-anchor" href="#步骤标（-step）" aria-hidden="true">#</a></h3><div class="language-"><pre><code>&lt;fx-step&gt;
  &lt;template #step=&quot;{ finish, index, active }&quot;&gt;
    &lt;fx-icon v-if=&quot;finish&quot; icon=&quot;CheckOutlined&quot;&gt;&lt;/fx-icon&gt;
  &lt;/template&gt;
&lt;/fx-step&gt;
</code></pre></div><p>PS：只推荐写入 text 和 <a href="./Icon.html">Icon</a> 组件，其他元素或组件可能会导致未定义的行为。</p></div></div><footer class="page-footer" data-v-7eddb2c4 data-v-fb8d84c6><div class="edit" data-v-fb8d84c6><div class="edit-link" data-v-fb8d84c6 data-v-1ed99556><!----></div></div><div class="updated" data-v-fb8d84c6><!----></div></footer><div class="next-and-prev-link" data-v-7eddb2c4 data-v-38ede35f><div class="container" data-v-38ede35f><div class="prev" data-v-38ede35f><a class="link" href="/vfox/components/CountDown" data-v-38ede35f><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-38ede35f><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-38ede35f>CountDown</span></a></div><div class="next" data-v-38ede35f><a class="link" href="/vfox/components/Price" data-v-38ede35f><span class="text" data-v-38ede35f>Price</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-38ede35f><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"apis_actionsheet.md\":\"12a2facc\",\"apis_calendar.md\":\"f3a21ae4\",\"apis_cascader.md\":\"b32af542\",\"apis_datepicker.md\":\"f01f569c\",\"apis_dialog.md\":\"d1994395\",\"apis_notify.md\":\"7e8dcabb\",\"apis_picker.md\":\"727671ce\",\"apis_popdialog.md\":\"8e7a722d\",\"apis_popmenu.md\":\"0a4ff1db\",\"apis_popover.md\":\"260f32ec\",\"apis_preview.md\":\"fe7aa48c\",\"apis_toast.md\":\"372f861c\",\"apis_index.md\":\"e8051b9c\",\"components_actionsheet.md\":\"f36d3d19\",\"components_activityindicator.md\":\"f3f7b345\",\"components_avatar.md\":\"91348fa4\",\"components_backtop.md\":\"00bbe717\",\"components_badge.md\":\"08565e6a\",\"components_button.md\":\"4fc09c4d\",\"components_calendar.md\":\"f2745ae9\",\"components_cascader.md\":\"10fa8a9c\",\"components_cell.md\":\"eac6d134\",\"components_checkbox.md\":\"be1a5a09\",\"components_circleprogress.md\":\"4cc69617\",\"components_collapse.md\":\"78a020b2\",\"components_copy.md\":\"ae13d133\",\"components_countdown.md\":\"4d61e8dd\",\"components_countup.md\":\"ff9c9b6a\",\"components_datepicker.md\":\"6598470c\",\"components_dialog.md\":\"94cdd590\",\"components_divider.md\":\"7ddb975e\",\"components_drawer.md\":\"d1fad034\",\"components_dropdown.md\":\"5b66d30b\",\"components_empty.md\":\"0baed8a3\",\"components_fixed.md\":\"eb55cbee\",\"components_flatlist.md\":\"c7c8ff8b\",\"components_form.md\":\"e61ae43d\",\"components_group.md\":\"dfe833c8\",\"components_icon.md\":\"9135be4a\",\"components_image.md\":\"d3b4af70\",\"components_imagepreview.md\":\"307ab8fc\",\"components_imageuploader.md\":\"6f908687\",\"components_indexview.md\":\"561e92d6\",\"components_input.md\":\"9d677004\",\"components_layout.md\":\"74180226\",\"components_loadmore.md\":\"b5b0c7a8\",\"components_modal.md\":\"08e61bf0\",\"components_navbar.md\":\"bf9d2bbc\",\"components_noticebar.md\":\"d3af742b\",\"components_notify.md\":\"0ae4c068\",\"components_numberkeyboard.md\":\"ce577764\",\"components_order.md\":\"6ef6afb2\",\"components_pagination.md\":\"a714d185\",\"components_picker.md\":\"306bfa4b\",\"components_popdialog.md\":\"1965a49b\",\"components_popmenu.md\":\"7199ab9d\",\"components_popover.md\":\"eab58dc4\",\"components_price.md\":\"59c63109\",\"components_progress.md\":\"e6c7a8db\",\"components_radio.md\":\"9b1b0c2a\",\"components_range.md\":\"24f0dec8\",\"components_rate.md\":\"cee7d045\",\"components_result.md\":\"ceaed6c7\",\"components_scrolltab.md\":\"5485a1c3\",\"components_scrollview.md\":\"f1cb6b18\",\"components_searchbar.md\":\"a16025c5\",\"components_sidetab.md\":\"5344ce2e\",\"components_skeleton.md\":\"991e17a1\",\"components_slider.md\":\"51f81a20\",\"components_stepper.md\":\"81dfce56\",\"components_steps.md\":\"9ca726e6\",\"components_sticky.md\":\"0ea9cb77\",\"components_stopwatch.md\":\"820d7c97\",\"components_swipecell.md\":\"aae3b380\",\"components_swiper.md\":\"0e52e650\",\"components_switch.md\":\"d0649c0e\",\"components_tab.md\":\"7fd3f7cd\",\"components_tabbar.md\":\"2fb76721\",\"components_tabview.md\":\"84505aa6\",\"components_tag.md\":\"e931e26c\",\"components_timeago.md\":\"f9e44191\",\"components_timeline.md\":\"df2c8269\",\"components_toast.md\":\"59ecc67f\",\"components_index.md\":\"c4faec27\",\"design_color.md\":\"e96819b3\",\"design_index.md\":\"946503cb\",\"guide_dark.md\":\"89d5b423\",\"guide_import.md\":\"1f0c7034\",\"guide_locale.md\":\"364f9493\",\"guide_theme.md\":\"744ae25d\",\"index.md\":\"1c3b2724\"}")</script>
    <script type="module" async src="/vfox/assets/app.9dc78301.js"></script>
    
  </body>
</html>