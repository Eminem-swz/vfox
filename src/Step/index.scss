@import '../style/var.scss';

.fx-step {
  --step-color: var(--steps-color);
  --step-active-color: var(--steps-active-color);

  position: relative;

  + .fx-step {
    &::before {
      content: '';
      margin-left: 40px;
    }
  }

  &_index {
    position: absolute;
    left: 16px;
    top: 15px;
    width: 18px;
    height: 18px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: 1px solid var(--step-color);
    box-sizing: border-box;
    border-radius: 50%;
    color: var(--step-color);
    background-color: #fff;
    font-size: 10px;
    overflow: hidden;
  }

  &_icon {
    --size: 12px;
    --color: var(--step-color);
  }

  &_line {
    position: absolute;
    left: 24px;
    top: 24px;
    width: 2px;
    height: 100%;
    background-color: var(--step-color);
  }

  &_inner {
    padding: 12px 16px 12px 40px;
  }

  &_title {
    font-size: 17px;
    line-height: 24px;
    word-break: break-all;
    word-wrap: break-word;
    overflow: hidden;
    font-weight: 500;
    color: $font-color;

    + .fx-step_content {
      margin-top: 4px;
    }

    a {
      text-decoration: none;
      color: $primary-color;
    }
  }

  &_content {
    font-size: 14px;
    line-height: 20px;
    word-break: break-all;
    word-wrap: break-word;
    overflow: hidden;
    color: $font2-color;

    a {
      text-decoration: none;
      color: $primary-color;
    }
  }

  &:last-child {
    .fx-step_line {
      display: none;
    }
  }

  &.finish {
    .fx-step {
      &_line {
        background-color: var(--step-active-color);
      }

      &_index {
        border-color: var(--step-active-color);
        color: var(--step-active-color);
      }

      &_icon {
        --color: var(--step-active-color);
      }
    }
  }

  &.active {
    .fx-step {
      &_line {
        background-color: var(--step-color);
      }

      &_index {
        border-color: var(--step-active-color);
        background-color: var(--step-active-color);
        color: #fff;
      }

      &_icon {
        --color: #fff;
      }
    }
  }
}
