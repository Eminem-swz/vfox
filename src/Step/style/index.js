import '../../style/common'
import '../../Steps/style/index'
