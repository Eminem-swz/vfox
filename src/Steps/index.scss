@import '../style/var.scss';

.fx-steps {
  --steps-color: #{$font3-color};
  --steps-active-color: #{$primary-color};

  &.dot .fx-step_index:not(.has--icon) {
    width: 8px;
    height: 8px;
    border-width: 2px;
    text-indent: -9999px;
    left: 21px;
    top: 20px;
  }
}
