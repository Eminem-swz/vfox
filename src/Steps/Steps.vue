<template>
  <div class="fx-steps" :class="{ dot, horizontal }" ref="list">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { stepsEmits, useStepList } from '@/Steps/steps'

export default defineComponent({
  name: 'fx-steps',
  props: {
    activeIndex: {
      type: Number,
      default: 0
    },
    dot: {
      type: Boolean,
      default: false
    },
    horizontal: {
      type: Boolean,
      default: false
    }
  },
  emits: [...stepsEmits],
  setup(props, ctx) {
    return { ...useStepList(props, ctx, 'steps') }
  }
})
</script>
