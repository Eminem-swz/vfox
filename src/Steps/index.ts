import { App } from 'vue'
import { SFCWithInstall } from '@/helpers/types'
import Steps from './Steps.vue'

const _Steps: SFCWithInstall<typeof Steps> = Object.assign(Steps, {
  install: function (app: App) {
    app.component(Steps.name, Steps)
  }
})

export default _Steps
