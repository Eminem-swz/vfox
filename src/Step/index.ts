import { App } from 'vue'
import { SFCWithInstall } from '@/helpers/types'
import Step from '../Steps/Step.vue'

const _Step: SFCWithInstall<typeof Step> = Object.assign(Step, {
  install: function (app: App) {
    app.component(Step.name, Step)
  }
})

export default _Step
