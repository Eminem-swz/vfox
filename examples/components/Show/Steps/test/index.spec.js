/* global describe,test,expect */

import { shallowMount } from '@vue/test-utils'
import Steps from '@/Steps'
import Step from '@/Step'

describe('Steps', () => {
  test('Steps should render default correctly', () => {
    const wrapper = shallowMount(Steps, {})

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('Step should render default correctly', () => {
    const wrapper = shallowMount(Step, {
      props: { title: 'title-1', content: 'content-1' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
