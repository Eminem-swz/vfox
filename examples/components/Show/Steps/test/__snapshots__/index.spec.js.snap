// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps Step should render default correctly 1`] = `
<div class="fx-step fx-steps-item fx-horizontal-hairline finish">
  <div class="fx-step_line"></div>
  <div class="fx-step_index"><span>0</span></div>
  <div class="fx-step_inner">
    <div class="fx-step_title">title-1</div>
    <div class="fx-step_content">content-1</div>
  </div>
</div>
`;

exports[`Steps Steps should render default correctly 1`] = `<div class="fx-steps"></div>`;
